clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp

```diff
@@ -54,6 +54,9 @@ void tooling::dependencies::configureInvocationForCaching(
       PPOpts.MacroIncludes.clear();
       PPOpts.Includes.clear();
     }
+    // Disable `-gmodules` to avoid debug info referencing a non-existent PCH
+    // filename.
+    CI.getCodeGenOpts().DebugTypeExtRefs = false;
```

Comment:
Do you think we should also reset `getHeaderSearchOpts().ModuleFormat` to "raw"? Technically it probably doesn't matter, but it's consistent with -go-modules which controls both options.

Reply:
Code already resets `HeaderSearchOptions`; I've added a note that we'd have to preserve `HeaderSearchOptions::ModuleFormat` when properly supporting `-gmodules`.